junit-vintage-engine/src/test/java/org/junit/vintage/engine/VintageTestEngineExecutionTests.java

```diff
@@ -683,12 +685,56 @@ void executesCompletelyDynamicTestCaseDiscoveredByUniqueId() {
 			event(engine(), finishedSuccessfully()));
 	}
 
-	private static Events execute(Class<?> testClass) {
+	@Test
+	void executesJUnit3ParallelSuiteWithSubsuites() {
+		var suiteClass = JUnit3ParallelSuiteWithSubsuites.class;
+		var results = execute(suiteClass);
+		results.containerEvents() //
+				.assertStatistics(stats -> stats.started(4).dynamicallyRegistered(0).finished(4).succeeded(4)) //
+				.assertEventsMatchExactly( //
+					event(engine(), started()), //
+					event(container(suiteClass), started()), //
+					event(container("Case"), started()), //
+					event(container("Case")), //
```

Comment:
Can this be Case1/Case2?
Are the events distinguishable?

Reply:
Since they run concurrently the order isn't reliable.

Reply:
Does it make sense to add a second test where regular `TestSuite` is used instead of `ActiveSuite`?
It should reproduce the same issue, however, the order would be reliable.